ambry-store/src/test/java/com.github.ambry.store/StorageManagerTest.java

```diff
@@ -209,16 +240,28 @@ private static StorageManager createAndStartStoreManager(List<ReplicaId> replica
    * @param storageManager the {@link StorageManager} to shutdown.
    * @param replicas the {@link ReplicaId}s to check for store inaccessibility.
    * @throws StoreException
+   * @throws InterruptedException
    */
   private static void shutdownAndAssertStoresInaccessible(StorageManager storageManager, List<ReplicaId> replicas)
```

Comment:
Could you add an `expectedStoreShutdownFailureCount` argument to this method? You could then do the metric check in this function.